apps/src/code-studio/components/libraries/LibraryPublisher.jsx

```diff
@@ -0,0 +1,248 @@
+    });
+
+    if (!(libraryDescription && functionsToPublish.length > 0)) {
+      this.setState({publishState: PublishState.INVALID_INPUT});
+      return;
+    }
+
+    let libraryJson = libraryParser.createLibraryJson(
+      librarySource,
+      functionsToPublish,
+      libraryName,
+      libraryDescription
+    );
+
+    this.props.clientApi.publish(
+      libraryJson,
+      error => {
+        console.warn(`Error publishing library: ${error}`);
+        this.setState({publishState: PublishState.ERROR_PUBLISH});
+      },
+      () => {
+        this.props.onPublishSuccess(this.state.libraryName);
+      }
+    );
+    dashboard.project.setLibraryName(this.state.libraryName);
+    dashboard.project.setLibraryDescription(libraryDescription);
+  };
+
+  displayNameInput = () => {
+    return (
+      <div>
+        <input
+          style={styles.textInput}
+          type="text"
+          value={this.state.libraryName}
+          onChange={this.setLibraryName}
+          onBlur={event =>
+            this.setState({
+              libraryName: libraryParser.suggestName(event.target.value)
+            })
+          }
+        />
+        <div style={styles.info}>{i18n.libraryNameRequirements()}</div>
+      </div>
+    );
+  };
+
+  resetErrorMessage = () => {
+    if (
+      this.state.libraryDescription &&
+      Object.values(this.state.selectedFunctions).find(value => value) &&
+      this.state.publishState === PublishState.INVALID_INPUT
+    ) {
+      this.setState({publishState: PublishState.DEFAULT});
+    }
+  };
+
+  displayDescription = () => {
+    return (
+      <textarea
+        rows="2"
+        cols="200"
+        style={{...styles.textInput, ...styles.description}}
+        placeholder={i18n.libraryDescriptionPlaceholder()}
+        value={this.state.libraryDescription}
+        onChange={event =>
+          this.setState({libraryDescription: event.target.value})
+        }
+        onBlur={this.resetErrorMessage}
+      />
+    );
+  };
+
+  boxChecked = name => () => {
+    this.setState(state => {
+      state.selectedFunctions[name] = !state.selectedFunctions[name];
+      return state;
+    }, this.resetErrorMessage);
+  };
+
+  displayFunctions = () => {
+    return this.props.libraryDetails.sourceFunctionList.map(sourceFunction => {
+      let name = sourceFunction.functionName;
+      let comment = sourceFunction.comment;
+      let shouldDisable = comment.length === 0;
+      let checked = this.state.selectedFunctions[name] || false;
+      if (shouldDisable && checked) {
+        checked = false;
+        this.setState(state => {
+          state.selectedFunctions[name] = false;
+          return state;
+        });
+      }
+      return (
+        <div key={name}>
+          <input
+            style={styles.largerCheckbox}
+            type="checkbox"
+            disabled={shouldDisable}
+            name={name}
+            checked={checked}
+            onChange={this.boxChecked(name)}
```

Comment:
i think this event handler should be wrapped in a function, right?

```suggestion
            onChange={() => this.boxChecked(name)}
```

Reply:
Woops. Thanks for pointing that out. Turns out I was accidentally using an uncommon method for handling this - I had made `boxChecked` into a double arrow function. Fixing that in the next commit.